### ISSUE TITLE
`// swiftlint:disable:next` does not work if there are doc comments

### ISSUE THREAD

**Issue by wdg** [realm/SwiftLint#4032]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`// swiftlint:disable:next` does not disable as expected.

Expectation
```swift
// swiftlint:disable:next cyclomatic_complexity
/// Returns a `Color` for a specific `fileType`
/// If not specified otherwise this will return `Color.accentColor`
public static func iconColor(fileType: FileType) -> Color {
```
Reality: SwiftLint triggers warning about cyclomatic_complexity

This works (kinda, but triggers another warning) 
```swift
/// Returns a `Color` for a specific `fileType`
/// If not specified otherwise this will return `Color.accentColor`
// swiftlint:disable:next cyclomatic_complexity
public static func iconColor(fileType: FileType) -> Color {
```
Triggers warning: `Orphaned Doc Comment Violation: A doc comment should be attached to a declaration. (orphaned_doc_comment)`

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --strict
... SNAP ...
/Users/wesley/Code/Apps/MacOS/*/*/*/*/src/UI/FileIcon.swift:47:5: error: Orphaned Doc Comment Violation: A doc comment should be attached to a declaration. (orphaned_doc_comment)
/Users/wesley/Code/Apps/MacOS/*/*/*/*/src/UI/FileIcon.swift:48:5: error: Orphaned Doc Comment Violation: A doc comment should be attached to a declaration. (orphaned_doc_comment)
/Users/wesley/Code/Apps/MacOS/*/*/*/*/src/UI/FileIcon.swift:99:5: error: Orphaned Doc Comment Violation: A doc comment should be attached to a declaration. (orphaned_doc_comment)
/Users/wesley/Code/Apps/MacOS/*/*/*/*/src/UI/FileIcon.swift:100:5: error: Orphaned Doc Comment Violation: A doc comment should be attached to a declaration. (orphaned_doc_comment)
Done linting! Found 17 violations, 17 serious in 302 files.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.47.1

* Installation method used (Homebrew, CocoaPods, building from source, etcetera)?
Homebrew

* Paste your configuration file:

```yml

# Disabled rule, reason.
disabled_rules:
  - todo # New project, we have a lot of // TODO:
  - missing_docs

# Exclude triggering type names.
type_name:
  excluded:
    - ID

# Exclude triggering identifier names.
identifier_name:
  excluded:
    - id
    - vc
    # (short) File extensions:
    - c
    - m
    - h
    - js
    - md
    - py
    - go
    - ts
    - txt
    - sh
    - pf
    - r
    - q
    - tp
    - xl
    - hy
    - d
    - cs

# Opt in rules, we want it more stricter.
opt_in_rules:
  - empty_count
  - closure_spacing
  - contains_over_first_not_nil
  - missing_docs
  - modifier_order
  - convenience_type
  - pattern_matching_keywords
  - modifier_order
  - identical_operands
  - empty_count
  - empty_string

# Custom configuration for nesting, this needs to be removed at some point.
nesting:
  type_level:
    warning: 2 # warning if you nest 2 level deep instead of 1
    error: 3 # error if you nest 3 level deep instead of 1

# Prefer spaces over tabs.
custom_rules:
  spaces_over_tabs:
    included: ".*\\.swift"
    name: "Spaces over Tabs"
    regex: "\t"
    message: "Prefer spaces for indents over tabs. See Xcode setting: 'Text Editing' -> 'Indentation'"
    severity: warning

  at_attributes_newlining:
    name: "Significant attributes"
    message: "Significant @attributes should be on an extra line"
    included: ".*.swift"
    regex: '(@objc\([^\)]+\)|@nonobjc|@discardableResult|@propertyWrapper|@UIApplicationMain|@dynamicMemberLookup|@_cdecl\([^\)]+\))[^\n]'
    severity: error

  multiple_empty_lines:
    included: ".*.swift"
    name: "Multiple Empty Lines"
    regex: '((?:\s*\n){3,})'
    message: "There are too many line breaks"
    severity: error

  comma_space_rule:
    regex: ",[ ]{2,}"
    message: "Expected only one space after ',"
    severity: error

  # Disable usage of // swiftlint:disable (rulename)
  swiftlint_file_disabling:
    included: ".*.swift"
    name: "SwiftLint File Disabling"
    regex: "swiftlint:disable\\s"
    match_kinds: comment
    message: "Prefer swiftlint:disable:next or swiftlint:disable:this"
    severity: warning

  # Disable Xcode placeholders like <#Description#>
  no_placeholders:
    included: ".*.swift"
    name: "No Placeholders"
    regex: "\\<#([a-zA-Z]+)#\\>"
    match_kinds: comment
    message: "Please do not use Xcode's placeholders."
    severity: warning
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
Nope

* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 14.0
Build version 14A5270f

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.
See examples

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by wdg** [comment]
Update

```swift
    static func `default`( 
        fileManager: FileManager,
        folderURL: URL,
        ignoredFilesAndFolders: [String],
        onUpdate: @escaping () -> Void = {}
    ) throws -> Self { // swiftlint:disable:this function_body_length
```
Does not work

```swift
    static func `default`( // swiftlint:disable:this function_body_length
        fileManager: FileManager,
        folderURL: URL,
        ignoredFilesAndFolders: [String],
        onUpdate: @escaping () -> Void = {}
    ) throws -> Self {
```
works..

**Comment by marcelofabri** [comment]
This expected. `next`, `previous` and `this` refer to lines only. So you need to use one of them based on where the violation is triggered/how you want your code to look.

Alternatively, you can use the "block" version:

```swift

// swiftlint:disable function_body_length

// <your function>

// swiftlint:enable function_body_length
```

**Comment by wti** [comment]
For readers finding this issue because there seems to be no way with complex types or functions to satisfy all of:

- Use single-line swiftlint:disable rather than scattered, multi-part block disable
- next: n/a b/c doc comment immediately before
- this: n/a b/c line length limited (esp. if restricted to 80)
- previous: n/a after multi-line declaration of parameters

one solution is to use `previous` just after the function name, before parameters:

```{swift}
/// doc comment
public static func someComplicatedFunction<T, U: Comparable>(
      // swiftlint:disable:previous function_body_length
      tool: T,
      _ arguments: [U] = [],
      input: Data = Data(),
      completionHandler: @escaping CompletionHandler
    ) { ... }
```

**Comment by SimplyDanny** [comment]
While people might interpret them as being applicable to the next/the previous/this violation, they refer to the next/the previous/this line. The keywords are kept short for ergonomic reasons, but the meaning is explained in the README:

> It's also possible to modify a disable or enable command by appending `:previous`, `:this` or `:next` for only applying the command to the previous, this (current) or next **line** respectively.

I don't see why this is a bug or should be changed in any way. Read once, remember forever. 😉

**Comment by mildm8nnered** [comment]
So I have lived with this, and I also did some work on SwiftLint disable processing in the codebase, and it is mildly annoying.

Personally I just use `this` or `previous`, which for some reason is less aesthetically pleasing, but works ok.

I think having `next` ignore comments is going to be painful, because there could be a violation in the comments.

For example, imagine a rule which said doc comments must end in a full stop. That might even be a thing already. So now the `swiftlint:disable` command is going to have to work out "am I supposed to be ignoring errors in the next line, or in the first line of code". And as rules don't "know" whether they apply to comments or not, there is no way for them to tell which is which.

**Comment by richardtop** [comment]
Doesn't work, this should be reopened.

**Comment by SimplyDanny** [comment]
> Doesn't work, this should be reopened.

Be more precise please.
